mapwms.c

```diff
@@ -3670,6 +3725,49 @@ int msWMSGetMap(mapObj *map, int nVersion, char **names, char **values, int nume
 
   } else
     img = msDrawMap(map, MS_FALSE);
+
+  /* see if we have tiled = true and a buffer */
+  /* if so, clip the image */
+  for (i=0; i<numentries; i++) {
+    if (strcasecmp(names[i], "TILED") == 0) {
+      if (strcasecmp(values[i], "TRUE") == 0) {
+        rendererVTableObj *renderer;
+        hashTableObj *meta = &(map->web.metadata);
+        rasterBufferObj imgBuffer;
+        imageObj *tmp;
+        const char *value;
+        int width = 0;
+        int height = 0;
+
+        if ((value = msLookupHashTable(meta, "tile_map_edge_buffer")) != NULL) {
```

Comment:
It might be necessary to add some more error handling in this section. I didn't want to double-check too much.